### PR TITLE
bug fix: profile command should run when datasource not provided

### PR DESCRIPTION
**Pull request by eugmandel**

### CODE REVIEW

File: great_expectations/cli/cli.py [hunk 1]
```diff
@@ -239,9 +239,6 @@ def profile(datasource_name, data_assets, profile_all_data_assets, directory):
     DATASOURCE_NAME: the datasource to profile, or leave blank to profile all datasources."""
 
 
-    if profile_all_data_assets:
```
Comment (abegong) [inline great_expectations/cli/cli.py:242]: Why remove this? I think we want to keep this functionality.

File: great_expectations/cli/cli.py [hunk 2]
```diff
@@ -250,8 +247,11 @@ def profile(datasource_name, data_assets, profile_all_data_assets, directory):
 
     if datasource_name is None:
         datasources = [datasource["name"] for datasource in context.list_datasources()]
-        for datasource_name in datasources:
-            profile_datasource(context, datasource_name, max_data_assets=max_data_assets)
+        if len(datasources) > 1:
```
Comment (abegong) [inline great_expectations/cli/cli.py:259]: I can't quite tell how this is supposed to work with other variables.

Unit tests as examples would be **super helpful** here.

**Review by abegong — CHANGES_REQUESTED** [review]

File: great_expectations/cli/cli.py (hunk 2 above)
Comment (eugmandel) [reply great_expectations/cli/cli.py:259]: @abegong Added tests with/without/with valid/invalid args and a method doc.

**Review by abegong — APPROVED** [review]
LGTM. There's clearly a lot more we could test, but I feel better since we have at least a handful of tests in place.

If I were doing one thing next, it would be to verify that profiling actually produces output files. (Currently we're only testing CLI screen output.)

**Pull request merged**